src/bleanser/modules/iter.py

```diff
@@ -0,0 +1,58 @@
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Iterator, Any
+
+
+from bleanser.core.processor import BaseNormaliser
+
+
+class IterNormaliser(BaseNormaliser):
+    """
+    This is meant to be overridden by a subclass
+
+    parse_file receives an input file, and should yield data/objects that when converted
+    to a string, produces some comparable data/object to the cleaned output file
+
+    possible things this could return is a unique key/id, or a tuple of (key, data), or a
+    namedtuple/dataclass
+
+    newlines are stripped from the string, so lines can be compared/diffed properly
+
+    Its possible you could use a library or code from https://github.com/karlicoss/HPI
+    in parse_file, to use the DAL itself to parse the file https://beepb00p.xyz/exports.html#dal
+    """
+
+    def parse_file(self, path: Path) -> Iterator[Any]:
```

Comment:
Maybe a better name would be `ExtractObjectsNormaliser` and `extract_objects` instead of `parse_file` (or something like that)? Feels like it would represent exactly what's happening here?